### DISCUSSION TITLE
MyST-NB "Cell-Tag configuration" with Sphinx-Design Dropdown instead of sphinx-togglebutton

### DISCUSSION THREAD

**Discussion by kai-tub** [executablebooks/meta#778] [Q&A]
Hey, I have a 'high-level' question where I am having trouble following how all the things work together.
I like the feature of `sphinx-togglebutton` that allows me to configure the tag of a Jupyter notebook cell and write `tags = ["hide-input"]`  to _hide_ the cells input behind a button with which the user can interact.

Now I would like to use the toggle functionality from the Sphinx-Design's Dropdown implementation instead because it is very good-looking, configurable, no JS dependencies, etc.

I have a basic understanding of the docutils tree/nodes implementation but I cannot find the exact point where tag `hide-input` is read from (presumably?) MyST-NB and where it chooses to _add_ the sphinx-togglebutton node.

I tried searching it from within GitHub across the entire organization but I can't seem to find the point where this is added.
Or am I misunderstanding something?

Thank you for your help!

PS: I am loving the work of the executablebooks project! And by the end of this week, I might have something to give back to the project :)

**Comment by kai-tub** [comment]
After looking through how the `render.py` file is structured, I guess the best solution would be to define a SphinxPostTransform that looks inside of the node containers for the set metadata (which can then be customized with new names):

https://github.com/executablebooks/MyST-NB/blob/876edf37f174369afce3738b9b26371864dc03a2/myst_nb/core/render.py#L163-L190

And then update the doc-tree structure.
Ok, maybe not ShpinxPostTransform but something similar.

Would this be something that should be contributed directly to the `sphinx-design` project?

EDIT: The main disadvantage is that we would be coupling internal behavior. sphinx-design would use the _knowledge_ that metadata is injected into the source code cell nodes and the added code would only be useful for the myst-nb library.

**Comment by choldgraf** [comment]
I think one nice solution would be to re-use sphinx design in Myst-nb instead of sphinx-togglebutton, then we could deprecate the "toggle directive" functionality in sphinx toggle button and define some more custom CSS for cell toggles in the themes.

One note - an upcoming release of sphinx toggle button is likely going to make the visual style of the toggle buttons to be more similar in structure to sphinx design, though with a bit less ink

- https://github.com/executablebooks/sphinx-togglebutton/pull/47

**Reply by kai-tub** [reply]
Thank you for the prompt reply!

> I think one nice solution would be to re-use sphinx design in Myst-nb instead of sphinx-togglebutton, then we could deprecate the "toggle directive" functionality in sphinx toggle button and define some more custom CSS for cell toggles in the themes.

I have a couple of follow up questions:

> I think one nice solution would be to re-use sphinx design in Myst-nb instead of sphinx-togglebutton

So by re-use do you mean _making_ mystnb compatible with the sphinx-design directive?
Like extending the functionality of mystnb?

Or adding code to sphinx-design to make it possible to _only_ install sphinx-design in combination with mystnb and have them 'working' together, which would require sphinx-design to depend on the internal parsing process.

Or do you think this is work that doesn't fit with the current objective and is hard to let an _external_ contributor (me) work on it because it would take a lot of time discussing the different parts and reviewing the changes?
